graylog2-server/src/test/java/org/graylog2/indexer/rotation/strategies/SmartRotationStrategyTest.java

```diff
@@ -0,0 +1,115 @@
+import static org.graylog2.indexer.rotation.strategies.SmartRotationStrategy.MIN_INDEX_SIZE;
+import static org.graylog2.indexer.rotation.strategies.SmartRotationStrategy.ROTATION_PERIOD;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class SmartRotationStrategyTest {
+
+    private SmartRotationStrategy smartRotationStrategy;
+
+    @Mock
+    private Indices indices;
+
+    @Mock
+    private NodeId nodeId;
+
+    @Mock
+    private AuditEventSender auditEventSender;
+
+    @Mock
+    private IndexSet indexSet;
+
+    private SmartRotationStrategyConfig smartRotationStrategyConfig;
+    private DateTime now;
+
+    @BeforeEach
+    void setUp() {
+        smartRotationStrategy = new SmartRotationStrategy(indices, nodeId, auditEventSender, new ElasticsearchConfiguration());
+
+        smartRotationStrategyConfig = SmartRotationStrategyConfig.builder().build();
+        final IndexSetConfig indexSetConfig = mock(IndexSetConfig.class);
+        when(indexSetConfig.rotationStrategy()).thenReturn(smartRotationStrategyConfig);
+        when(indexSet.getConfig()).thenReturn(indexSetConfig);
+        now = DateTime.now(DateTimeZone.UTC);
+    }
+
+    @Test
+    void shouldRotateWhenTooBig() {
+        final DateTime creationDate = now.minus(Duration.standardMinutes(10));
+        when(indices.indexCreationDate("index_0")).thenReturn(Optional.of(creationDate));
+        when(indices.getStoreSizeInBytes("index_0")).thenReturn(Optional.of(MAX_INDEX_SIZE.toBytes() + 10));
+
+        final SmartRotationStrategy.Result result = smartRotationStrategy.shouldRotate("index_0", indexSet);
+
+        assertThat(result.shouldRotate()).isEqualTo(true);
+        assertThat(result.getDescription()).contains("exceeds MAX_INDEX_SIZE");
+    }
+
+    @Test
+    void shouldRotateWhenRightSizedAndOverRotationPeriod() {
+        // TODO only works with multiple of days
+        final DateTime creationDate = now.minus(Duration.standardDays(ROTATION_PERIOD.getDays()));
+        when(indices.indexCreationDate("index_0")).thenReturn(Optional.of(creationDate));
+        when(indices.getStoreSizeInBytes("index_0")).thenReturn(Optional.of(MIN_INDEX_SIZE.toBytes() + 10));
+
+        final SmartRotationStrategy.Result result = smartRotationStrategy.shouldRotate("index_0", indexSet);
+
+        assertThat(result.shouldRotate()).isEqualTo(true);
+        assertThat(result.getDescription()).contains("Index is old enough");
+    }
+
+    @Test
+    void shouldRotateWhenBeyondLeeWay() {
+        final Period leeWay = smartRotationStrategyConfig.indexLifetimeHard().minus(smartRotationStrategyConfig.indexLifetimeSoft());
+        // TODO only works with multiple of days
+        final java.time.Duration leeWayDuration = java.time.Duration.ofDays(leeWay.getDays() + ROTATION_PERIOD.getDays());
+        final DateTime creationDate = now.minus(leeWayDuration.toMillis());
```

Comment:
`creationDate` is right at the limit. The test fails if there isn't any elapsed time between computing this date and checking the rotation criteria. Make the creationdate a little bit earlier.

Reply:
Already fixed - nice